### PR TITLE
Editor: Titles for images / Quill v2

### PR DESCRIPTION
**Pull request by AntonGluschuk**
Editor: Titles for images  implementation

**Issue link**

[Editor: research the way to have titles for images #211](https://github.com/ita-social-projects/dokazovi-fe/issues/211)

## Summary of issue

- [x]  Implemented titles for images in the Editor

## Summary of change

- [x]  React-Quill updated to version 2

### CODE REVIEW

File: src/lib/components/Editor/utilities.ts [hunk 1]
```diff
@@ -1,14 +1,45 @@
...
+    const captionElement = domNode.querySelector('figcaption');
+    const url = imageElement.getAttribute('src');
+    const caption = captionElement.innerHTML;
+    return { url, caption };
+  }
+}
+FigureBlot.blotName = 'figureBlock';
+FigureBlot.tagName = 'FIGURE';
+Quill.register({ 'blots/figureBlock': FigureBlot });
+
```
Comment (SergeOlabin) [inline src/lib/components/Editor/utilities.ts:39]: extract to the separate file e.g. Editor/Blots/FigureBlot

File: src/lib/utilities/API/imgurApi.ts [hunk 2]
```diff
@@ -22,3 +22,20 @@ export const postImage = (
...
+// Max 12,500 requests - allowed during 1 month
+// If the daily limit is hit five times in a month, then the app will be blocked for the rest of the month
+export const getLimits = (
+  config = {
+    headers: {
+      Authorization: `Client-ID ${CLIENT_ID}`,
+    },
+  },
+): Promise<AxiosResponse> => {
+  return axios.get(`${BASE_URL}/credits`, {
```
Comment (SergeOlabin) [inline src/lib/utilities/API/imgurApi.ts:37]: use axions instance provided above, you don't need to put headers explicetly here, there're intercetpors added

**Review by SergeOlabin — CHANGES_REQUESTED** [review]

File: src/lib/components/Editor/utilities.ts (hunk 1 above)
Comment (AntonGluschuk) [reply src/lib/components/Editor/utilities.ts]: fixed

File: src/lib/utilities/API/imgurApi.ts (hunk 2 above)
Comment (AntonGluschuk) [reply src/lib/utilities/API/imgurApi.ts:37]: fixed

**Review by SergeOlabin — APPROVED** [review]

**Review by DelFlumen — APPROVED** [review]

**Pull request merged**